kedro/framework/cli/starters.py

```diff
@@ -62,6 +72,34 @@ def _remove_readonly(func: Callable, path: Path, excinfo: Tuple):  # pragma: no
     func(path)
 
 
+def _get_starters_aliases() -> Dict[str, Dict[str, str]]:
+    """This functions lists all the starters aliases declared in
+    the core repo and in plugins entry points.
+    """
+    official_starter_aliases = _STARTER_ALIASES.copy()
+    starters_aliases_from_plugins = load_entry_points("starter")
+
+    # TODO: it would likely be a good idea to indicate in the name
+    # from which plugin this is imported, e.g  we may store the origin
+    # which would return a dict like {pandas-iris: kedro, my_awesome-starter: awesome_plugin}
+    # this is useful for debugging and when listing discovered starters
+
+    # TODO Add the possibility for plugins users to specify {starter_name: git+url}
+    #  instead of {starter_name: {"repo": git+url, "directory": None}}
+    # which is unneedly verbose and error prone
```

Comment:
Agree the user shouldn't need to specify `directory` (it should default to `None`) but not sure about shortcutting out the nested `repo` (or `template_path`) key. It's more verbose but I think clearer than introducing a new shortcut syntax.

Reply:
I agree for not shortcutting, it would add extra maitneane burden for very little better user experience